Nfc error when approaching a yubico 5 nfc

after updating to version 5.6 the encryption via NFC no longer works.  in practice, before I approached yubico Nfc to the cell phone and confirmation of encryption came out, now instead when it asks to approach the yubico via Nfc, it crashes or opens Chrome.  the control no longer applies at the time of the nfc

![Screenshot_20210215-184004](https://user-images.githubusercontent.com/35575744/107978627-693c6580-6fbd-11eb-849e-86a7fc970573.png)

![Screenshot_20210215-184028](https://user-images.githubusercontent.com/35575744/107978643-70637380-6fbd-11eb-940d-04e0df10ad2c.png)

found the (likely) crash report on the play store:

```
java.lang.RuntimeException: 
  at android.os.AsyncTask$4.done (AsyncTask.java:415)
  at java.util.concurrent.FutureTask.finishCompletion (FutureTask.java:383)
  at java.util.concurrent.FutureTask.setException (FutureTask.java:252)
  at java.util.concurrent.FutureTask.run (FutureTask.java:271)
  at android.os.AsyncTask$SerialExecutor$1.run (AsyncTask.java:305)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1167)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:641)
  at java.lang.Thread.run (Thread.java:923)
Caused by: java.lang.IllegalStateException: 
  at org.sufficientlysecure.keychain.securitytoken.AutoValue_KdfParameters$Builder.build (AutoValue_KdfParameters.java:21)
  at org.sufficientlysecure.keychain.securitytoken.KdfParameters.fromKdfDo (KdfParameters.java:2)
  at org.sufficientlysecure.keychain.securitytoken.SecurityTokenConnection.retrieveKdfDo (SecurityTokenConnection.java:7)
  at org.sufficientlysecure.keychain.securitytoken.SecurityTokenConnection.calculateKdfIfNecessary (SecurityTokenConnection.java:2)
  at org.sufficientlysecure.keychain.securitytoken.SecurityTokenConnection.verifyPinForOther (SecurityTokenConnection.java:4)
  at org.sufficientlysecure.keychain.securitytoken.operations.PsoDecryptTokenOp.verifyAndDecryptSessionKey (PsoDecryptTokenOp.java:1)
  at org.sufficientlysecure.keychain.ui.SecurityTokenOperationActivity.doSecurityTokenInBackground (SecurityTokenOperationActivity.java:57)
  at org.sufficientlysecure.keychain.ui.base.BaseSecurityTokenActivity.handleSecurityToken (BaseSecurityTokenActivity.java:1)
  at org.sufficientlysecure.keychain.ui.base.BaseSecurityTokenActivity$1.doInBackground (BaseSecurityTokenActivity.java:3)
  at org.sufficientlysecure.keychain.ui.base.BaseSecurityTokenActivity$1.doInBackground (BaseSecurityTokenActivity.java:1)
  at android.os.AsyncTask$3.call (AsyncTask.java:394)
  at java.util.concurrent.FutureTask.run (FutureTask.java:266)
```

(despite how this looks btw, version 5.6 still fixed a lot of crashes for folks. we cut the number of crash reports by half or so :)

--- a/OpenKeychain/src/main/java/org/sufficientlysecure/keychain/securitytoken/KdfParameters.java
+++ b/OpenKeychain/src/main/java/org/sufficientlysecure/keychain/securitytoken/KdfParameters.java
@@ -85,6 +85,7 @@ public abstract class KdfParameters {
                             case (byte)0x00:
                                 // no KDF, plain password
                                 hasUsesKdf(false);
+                                break;
                             case (byte)0x03:
                                 // using KDF
                                 hasUsesKdf(true);

